src/routes/safe/components/Transactions/TxList/TxOwners.tsx

```diff
@@ -24,92 +100,86 @@ export const TxOwners = ({
+      ? // Threshold reached
+        detailedExecutionInfo.confirmationsRequired - detailedExecutionInfo.confirmations.length <= 0 ||
+          // Less than 3 confirmations
+          detailedExecutionInfo.confirmations.length <= 3
+      : true,
+  )
+
+  const toggleHide = () => {
+    setHideConfirmations((prev) => !prev)
+  }
+
+  if (!detailedExecutionInfo || !isMultiSigExecutionDetails(detailedExecutionInfo)) {
     return null
   }
 
   const confirmationsNeeded = detailedExecutionInfo.confirmationsRequired - detailedExecutionInfo.confirmations.length
-
-  const CreationNode = isCancelTxDetails(txInfo) ? (
-    <OwnerListItem>
-      <span className="icon">
-        <Icon size="sm" type="circleCross" color="error" />
-      </span>
-      <div className="legend">
-        <Text color="error" size="xl" strong>
-          On-chain rejection created
-        </Text>
-      </div>
-    </OwnerListItem>
-  ) : (
-    <OwnerListItem>
-      <span className="icon">
-        <Icon size="sm" type="add" color="primary" />
-      </span>
-      <div className="legend">
-        <Text color="primary" size="xl" strong>
-          Created
-        </Text>
-      </div>
-    </OwnerListItem>
-  )
+  const isConfirmed = confirmationsNeeded <= 0
+  const isExecuted = !!detailedExecutionInfo.executor
 
   return (
-    <OwnerList>
-      {CreationNode}
-      {detailedExecutionInfo.confirmations.map(({ signer }) => (
-        <OwnerListItem key={signer.value}>
-          <span className="icon">
-            <Icon size="sm" type="circleCheck" color="primary" />
-          </span>
-          <div className="legend">
-            <Text color="primary" size="xl" strong>
-              Confirmed
-            </Text>
-            <AddressInfo
-              address={signer.value}
-              name={signer?.name || undefined}
-              avatarUrl={signer?.logoUri || undefined}
-              shortenHash={4}
-            />
-          </div>
-        </OwnerListItem>
-      ))}
-      {isPending || confirmationsNeeded <= 0 ? (
-        <OwnerListItem>
-          <span className="icon">
-            {detailedExecutionInfo.executor ? (
-              <Icon type="circleCheck" size="sm" color="primary" />
-            ) : (
-              <StyledImg alt="" src={TransactionListActive} />
-            )}
+    <StyledStepper orientation="vertical" nonLinear connector={<StyledStepConnector />}>
+      {isCancelTxDetails(txInfo) ? (
+        <StyledStep bold color="red">
+          <StepLabel icon={<TxRejectionIcon />}>On-chain rejection created</StepLabel>
+        </StyledStep>
+      ) : (
+        <StyledStep bold color="green">
+          <StepLabel icon={<TxCreationIcon />}>Created</StepLabel>
+        </StyledStep>
+      )}
+      <StyledStep bold color={isConfirmed ? 'green' : 'orange'}>
+        <StepLabel icon={isConfirmed ? <CheckIcon /> : <CircleIcon />}>
+          Confirmations{' '}
+          <span style={{ color: black300, fontWeight: 'normal' }}>
```

Comment:
This (and any other similar styles) could be defined outside of the component so as to memoize them and not trigger a re-rendering.